app/bundles/CoreBundle/Form/Type/ContentPreviewSettingsType.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Form\Type;
+
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\EmailBundle\Entity\Email;
+use Mautic\PageBundle\Entity\Page;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Intl\Intl;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Preview settings form used for pages and emails in detail view page.
+ */
+class ContentPreviewSettingsType extends AbstractType
+{
+    public const TYPE_EMAIL = 'email';
+    public const TYPE_PAGE  = 'page';
+
+    private const CHOICE_TYPE_TRANSLATION = 'translation';
+    private const CHOICE_TYPE_VARIANT     = 'variant';
+
+    private TranslatorInterface $translator;
+
+    private CorePermissions $security;
+
+    public function __construct(TranslatorInterface $translator, CorePermissions $security)
+    {
+        $this->translator = $translator;
+        $this->security   = $security;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $objectId              = $options['objectId'];
+        $translations          = $options['translations'];
+        $variants              = $options['variants'];
+
+        $this->addTranslationOrVariantChoicesElement($builder, self::CHOICE_TYPE_TRANSLATION, $translations, $objectId);
+        $this->addTranslationOrVariantChoicesElement($builder, self::CHOICE_TYPE_VARIANT, $variants, $objectId);
+
+        if ($this->security->isAdmin()
+            || (!$this->security->hasEntityAccess('lead:leads:viewown', 'lead:leads:viewother'))
```

Comment:
Based on content condition should be without `!`. In other case its hidden for not admin users